assets/js/googlesitekit/notifications/datastore/notifications.js

```diff
@@ -155,14 +172,95 @@ export const reducer = createReducer( ( state, { type, payload } ) => {
 			break;
 		}
 
+		case RECEIVE_QUEUED_NOTIFICATIONS: {
+			state.queuedNotifications = payload.queuedNotifications;
+			break;
+		}
+
 		default:
 			break;
 	}
 } );
 
-export const resolvers = {};
+export const resolvers = {
+	*getQueuedNotifications( viewContext ) {
+		const registry = yield Data.commonActions.getRegistry();
+
+		const notifications = registry
+			.select( CORE_NOTIFICATIONS )
+			.getNotifications();
+
+		const filteredNotifications = Object.values( notifications ).filter(
+			( notification ) => {
+				if ( ! notification.viewContexts.includes( viewContext ) ) {
+					return false;
+				}
+
+				if (
+					!! notification.isDismissible &&
+					registry
+						.select( CORE_NOTIFICATIONS )
+						.isNotificationDismissed( notification.id )
+				) {
+					return false;
+				}
+
+				return true;
+			}
+		);
+
+		const queuedNotifications = yield Data.commonActions.await(
+			Promise.all(
+				filteredNotifications.filter( ( notification ) => {
+					if (
+						typeof notification.checkRequirements === 'function'
+					) {
+						return notification.checkRequirements( registry );
+					}
+
+					return true;
+				} )
+			)
```

Comment:
`filteredNotifications.filter` will return an array of notification objects and it won't wait for `checkRequirements` functions to resolve because we need to use `filteredNotifications.map` to get promisses that will be properly awaited with `Promise.all`.

But in this case the `queuedNotifications` will contain `true`/`false` results (we need to change `return true;` to be `return Promise.resolve( true )`) which means that we will need to do an extra step to filter notifications using the resulting array:

```js
const checks = yield Data.commonActions.await(
	Promise.all(
		filteredNotifications.map( ( notification ) => {
			if (
				typeof notification.checkRequirements === 'function'
			) {
				return notification.checkRequirements( registry );
			}

			return Promise.resolve( true );
		} )
	)
);

const queuedNotifications = filteredNotifications.filter( ( _, i ) => !! checks[ i ] );
```

This should do the trick if i understand it right.